WordPress/gutenberg
# TreeSelect: remove margin overrides and add new opt-in prop

## What?



Added new opt-in prop `__nextHasNoMarginBottom` for usages of `TreeControl` in the Gutenberg codebase and removed margin overrides. 

## Why?


Part of this project: https://github.com/WordPress/gutenberg/issues/38730
The tl;dr is `BaseControl` has a `margin-bottom` which makes it difficult to reuse and results in inconsistent use. 

## How?


By removing margin overrides in the CSS and adding the prop `__nextHasNoMarginBottom`.

## Testing Instructions


### HierarchicalTermSelector

1. Edit a post
2. Look for 'Categories' in the block inspector
3. Click on 'Add New Category'
4. Verify the spacing 

| Before  | After |
| ------------- | ------------- |
|  <img width="284" alt="Screenshot 2023-01-27 at 3 35 31 PM" src="https://user-images.githubusercontent.com/35543432/215225507-0c61ffb4-34db-4775-8439-da1427f84f23.png"> | <img width="287" alt="Screenshot 2023-01-27 at 3 32 00 PM" src="https://user-images.githubusercontent.com/35543432/215225524-54bd94ec-d83f-4806-bbe8-1fb34020eec5.png"> |

### CategorySelect & AuthorSelect

I don't believe that `CategorySelect` shows up anywhere/matches any conditions based on [this part of the readme](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/query-controls/README.md#multiple-category-selector):

>The `QueryControls` component now supports multiple category selection, to replace the single category selection... To enable it use the component with the new props instead: `categorySuggestions` in place of `categoriesList` 

So to test it, the only way I found was to replace `categorySuggestions={ categorySuggestions }` with `categoriesList={ categories }` on line 343 in `packages/block-library/src/latest-posts/edit.js`: https://github.com/WordPress/gutenberg/blob/6a64cdfee23442ee201ea5039ae5c833d1946428/packages/block-library/src/latest-posts/edit.js#L343

Then:

1. Add the Latest Posts block 
2. Click on the settings icon in the block inspector
3. Look for 'Category' and 'Author' within the 'Sorting and filtering' panel
4. Ensure the space below those select elements is the same as before 

<img width="285" alt="Screenshot 2023-01-27 at 2 46 48 PM" src="https://user-images.githubusercontent.com/35543432/215225597-8716877f-d9cc-4b51-8be9-d6c4f452349e.png">
